common/src/scheduler/mod.rs

```diff
@@ -50,13 +48,13 @@ pub trait Scheduler {
     /// Storing costs per block.
     type CostsPerBlock: SchedulingCostsPerBlock<BlockNumber = Self::BlockNumber, Cost = Self::Cost>;
 
-    /// Block numbers, which have already passed,
+    /// The first block number, which have already passed,
     /// but still contain tasks to deal with.
     ///
     /// Used for checking if scheduler is able to process
     /// current block aimed tasks, or there are some
     /// incomplete job from previous blocks.
-    type MissedBlocks: ValueStorage<Value = Self::MissedBlocksCollection>;
+    type FirstMissingBlock: ValueStorage<Value = Self::BlockNumber>;
```

Comment:
I'd rename it into something like `BlockPtr` or `BlockIterationHead`, cause it represents not necessary "missed" blocks

Reply:
UPD: we keep it with None if we go up-to-date, but anyway current naming seems misleading for me